apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -2007,15 +2011,18 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
 
     with :ok <- Queries.set_pending_empty_cache(db_client, device_id, true),
          :ok <- VMQPlugin.disconnect("#{realm}/#{encoded_device_id}", true) do
+      new_state = set_device_disconnected(state, db_client, timestamp)
+
+      MessageTracker.ack_delivery(new_state.message_tracker, message_id)
```

Comment:
I think you're "double acking" the message: you ack it here and then you discard it at the outer level (every `ask_clean_session` call is followed by a `MessageTracker.discard` call).

I think this `ack_delivery` is not needed here (and, by extension, you don't actually need to pass `message_id` to `ask_clean_session`), or am I missing something?